### ISSUE TITLE
[FEATURE] Surrealdb support 

### ISSUE THREAD

**Issue by Celibioux** [juspay/hyperswitch#4550]
### Feature Description

Adding support for Surrealdb Support would be great as it's a great Multi Modèles Database written in Rust 

### Possible Implementation

Being able to choose between Postgres or Surrealdb as a Database 

### Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by lsampras** [comment]
Hi @Celibioux ,

Is there any specific requirement that you are trying to address via surrealdb, It will help us understand & prioritize this support appropriately,

I'd be willing to review (and mentor) this although it is a low priority for the maintainers of this project,
This involves more effort than is apparent,

1. Currently we do have a provision to provide a configurable storage interface but it still has ways to go before usability
2. Currently we are strongly coupled with diesel in terms of our structs where it needs to be split up from the actual structs
3. We don't have a clear solution on how we will be managing the migrations / CI checks required to maintain the surrealdb implementations, would love your input on this part as well.

There will be considerable refactor required before we start adding support for multiple databases.
